lib/cartopy/mpl/geoaxes.py

```diff
@@ -1720,36 +1720,47 @@ def _pcolormesh_patched(self, *args, **kwargs):
 
                     collection.set_array(pcolormesh_data.ravel())
 
-                    # now that the pcolormesh has masked the bad values,
-                    # create a pcolor with just those values that were masked
-                    if C_mask is not None:
-                        # remember to re-apply the original data mask
-                        pcolor_data = np.ma.array(C, mask=~mask | C_mask)
-                    else:
-                        pcolor_data = np.ma.array(C, mask=~mask)
-
-                    pts = coords.reshape((Ny, Nx, 2))
-                    if np.any(~pcolor_data.mask):
-                        # plot with slightly lower zorder to avoid odd issue
-                        # where the main plot is obscured
-                        zorder = collection.zorder - .1
-                        kwargs.pop('zorder', None)
-                        kwargs.setdefault('snap', False)
-                        pcolor_col = self.pcolor(pts[..., 0], pts[..., 1],
-                                                 pcolor_data, zorder=zorder,
-                                                 **kwargs)
-
-                        pcolor_col.set_cmap(cmap)
-                        pcolor_col.set_norm(norm)
-                        pcolor_col.set_clim(vmin, vmax)
-                        # scale the data according to the *original* data
-                        pcolor_col.norm.autoscale_None(C)
-
-                        # put the pcolor_col on the pcolormesh collection so
-                        # that if really necessary, users can do things post
-                        # this method
-                        collection._wrapped_collection_fix = pcolor_col
-
+                    # the part below in comment retrieves the masked
+                    # overlapping cells and plot them with pcolor and a
+                    # smaller zorder to hide them
+                    # but there are not always hidden
+                    # if one wants to retrieve those cells, split them at the
+                    # boundary and plot them, the code below could be
+                    # useful.
+
+#                    #############
+#
+#                    # now that the pcolormesh has masked the bad values,
+#                    # create a pcolor with just those values that were masked
+#                    if C_mask is not None:
+#                        # remember to re-apply the original data mask
+#                        pcolor_data = np.ma.array(C, mask=~mask | C_mask)
+#                    else:
+#                        pcolor_data = np.ma.array(C, mask=~mask)
+#
+#                    pts = coords.reshape((Ny, Nx, 2))
+#
+#                    if np.any(~pcolor_data.mask):
+#                        # plot with slightly lower zorder to avoid odd issue
+#                        # where the main plot is obscured
+#                        zorder = collection.zorder - .1
```

Comment:
Did you try making this a much lower zorder like -1, so it is below all other artists?

Reply:
I tried a much lower order like -1.
In that case, it doesn't always work so I tried an ever larger number, and it worked.
Now I think there is no guarantee that it will always hid the cell, not plotting them is a better guarantee.
I think the issue comes with masked data, like oceanographic data with a land mask. I would say that when this was written, pushing the "hidden cell" behind the good one, was good enough, but when there is no data on the land mask, then the cells behind still appear on the masked area, not obscuring the data but the mask, and still making a mess of the plot.

Making it a kwargs is a possible option.

The idea was that those cell had to be hidden, so let's hide them efficiently. I admit that I spend quite some time thinking about it and I am sure of one thing is that cartopy needs to get rid of the cells...